### ISSUE TITLE
Testportal na Microsoft Teams

### ISSUE THREAD

**Issue by Pr0moK** [skhyr/anihilator#11]
Przychodzę z zapytaniem lub też z propozycją ponieważ, zauważyłem że nauczyciele zaczynają robić testy przez testportal lecz na ms teams, więc kieruje pytanie czy wtyczka również działa na testportalu na ms teams (oczywiście w przeglądarce), jeśli nie to czy istnieje możliwość aby także działała na ms teams.

**Comment by skhyr** [comment]
Aktualnie nie działa ale raczej trywialne będzie uruchomienie tej funkcjonalności.
_do zrobienia w trybie szybkim_

**Comment by Pr0moK** [comment]
Jeśli można, to chciałbym się zapytać jak idą pracę nad tą funkcją.

**Comment by Voodens** [comment]
Tak teraz by się przydała opcja na Teams

**Comment by Miller991** [comment]
Tak teraz by się przydała opcja na Teams

**Comment by PetrusTryb** [comment]
No, byłoby fajnie, tylko nie wydaje mi się, żeby było możliwe instalowanie wtyczek do Teamsów.
Jedyne co do tej pory odkryłem to odpalenie DevTools przez ukryte menu (7 razy lewym + raz prawym przyciskiem) -> devtools -> select web contents -> webview i tam w zakładce Rendering zaznaczenie Emulate Focused Page.

PoC tutaj:
https://github.com/PetrusTryb/school-hacks/blob/main/testportal/README.pl-PL.md#obej%C5%9Bcie-zabezpiecze%C5%84

**Comment by Voodens** [comment]
> No, byłoby fajnie, tylko nie wydaje mi się, żeby było możliwe instalowanie wtyczek do Teamsów.
> Jedyne co do tej pory odkryłem to odpalenie DevTools przez ukryte menu (7 razy lewym + raz prawym przyciskiem) -> devtools -> select web contents -> webview i tam w zakładce Rendering zaznaczenie Emulate Focused Page.
> 
> PoC tutaj:
> https://github.com/PetrusTryb/school-hacks/blob/main/testportal/README.pl-PL.md#obej%C5%9Bcie-zabezpiecze%C5%84

Ogólnie to odkryłem pewien sposób na oszukanie TestPortal na teams, wystarczy dodać dwie linijki do 
manifest.json 

 "*://*.teams.microsoft.com/*",
          "*://*.teams.testportal.net/*"

A w pliku Popup.js zmienić 
 fetch('https://raw.githubusercontent.com/Voodens/skr/main/manifest.json')

I powinno działać

**Comment by skhyr** [comment]
@Voodens czyli po prostu dopisanie żeby wtyczka się aktywowała na innych domenach. Zrobiłem commita, jeśli masz możliwość a wnioskuje że masz to sprawdź proszę czy najnowsza wersja działa i daj znać.

**Comment by skhyr** [comment]
@PetrusTryb do teamsów jako aplikacja desktopowa raczej się nie da ale zawsze na sprawdzian można wejść przez przeglądarkę i już będzie

**Comment by Voodens** [comment]
> @Voodens czyli po prostu dopisanie żeby wtyczka się aktywowała na innych domenach. Zrobiłem commita, jeśli masz możliwość a wnioskuje że masz to sprawdź proszę czy najnowsza wersja działa i daj znać.

W manifest.json zapomniałeś dodać = * 

Powiino to wyglądać tak 
 "*://*.testportal.pl/*",
          "*://*.testportal.net/*",
          "*://*.teams.microsoft.com/*",
          "*://*.teams.testportal.net/*"

Dodaj * w manifest.json

**Comment by PetrusTryb** [comment]
> @PetrusTryb do teamsów jako aplikacja desktopowa raczej się nie da ale zawsze na sprawdzian można wejść przez przeglądarkę i już będzie

Tak, w sensie oczywiście można się pobawić jakimś proxy zawsze (wystarczy zmienić jeden parametr, który idzie w requeście POST przy przesyłaniu odpowiedzi), ale faktycznie wsm wystarczy teamsy w przeglądarce odpalić (nawet lepiej bo w przeglądarce można uprawnienia odebrać i wtedy jest cisza w mikrofonie).

**Comment by kantraksel** [comment]
To aplikacja elektron, można coś zrobić na wzór Powercorda. Wymaga tylko znajomości mechanizmów, jakie tam zachodzą

**Comment by skhyr** [comment]
Gdyby to było PWA sprawa sama się rozwiązuje, zabawa w grzebanie dostępu do aplikacji electronowej, choć może i możliwa, to raczej nie opłacalna. Szybciej i łatwiej dla każdego po prostu wejść przez przeglądarkę na tą jedną lekcję.

**Comment by matt-b8** [comment]
Da radę wyłączyć blokadę wracania się do poprzednich pytań?

**Comment by Turqusowy** [comment]
Witam czy jest jakiś nowy obecny sposób na użycie tego rozszerzenia na teście z tesportalu na teamsie?
